### PR TITLE
minor fixes to build with qt6

### PR DESCRIPTION
**Pull request by benoitdm-oslandia**
When building from scratch a Ubuntu 23.10 docker image, I need to add this fixes to make `cmake` happy!

[X] can be backported

cc @lbartoletti  @troopa81

### CODE REVIEW

File: cmake/FindPyQt6.py [hunk 1]
```diff
@@ -48,7 +48,7 @@
 print("pyqt_version_str:%s" % PyQt6.QtCore.PYQT_VERSION_STR)
 print("pyqt_mod_dir:%s" % os.path.join(cfg.default_mod_dir, "PyQt6"))
 print("pyqt_sip_dir:%s" % sip_dir)
-print("pyqt_sip_flags:%s" % PyQt6.QtCore.PYQT_CONFIGURATION['sip_flags'])
+print("pyqt_sip_flags: ")
```
Comment (troopa81) [inline cmake/FindPyQt6.py:51]: Why removing the printed value?

Comment (benoitdm-oslandia) [reply cmake/FindPyQt6.py:51]: There is a dedicated commit for this :sweat_smile:: `PYQT_CONFIGURATION` does not exist any more in Qt6

File: cmake/FindQsci.cmake [hunk 2]
```diff
@@ -42,7 +42,7 @@ ELSE(QSCI_MOD_VERSION_STR)
 
     FIND_FILE(_find_qsci_py FindQsci.py PATHS ${CMAKE_MODULE_PATH} NO_CMAKE_FIND_ROOT_PATH)
 
-    SET(QSCI_VER 5)
+    STRING(REGEX REPLACE "^Qt([0-9]+)$" "\\1" QSCI_VER ${QT_VERSION_BASE})
```
[the remark this answers is not in the document]
Comment (benoitdm-oslandia) [reply cmake/FindQsci.cmake:45]: Indeed! I will apply this changes

[the remark this answers is not in the document]
Comment (benoitdm-oslandia) [reply cmake/FindQsci.cmake]: done!

File: cmake/FindPyQt6.py (hunk 1 above)
Comment (troopa81) [reply cmake/FindPyQt6.py:51]: Can we remove the print so ?

Comment (troopa81) [reply cmake/FindPyQt6.py:51]: as described [here](https://www.riverbankcomputing.com/static/Docs/PyQt6/pyqt5_differences.html) PYQT_CONFIGURATION has indeed been removed. I would be in favor to remove the print() and the subsequent modification in cmake file

**Review by troopa81 — APPROVED** [review]

**Pull request merged**